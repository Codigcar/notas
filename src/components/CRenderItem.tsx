import React from 'react';
import {Text, View, StyleSheet} from 'react-native';
import Proptypes from 'prop-types';

import {IPublication} from '../interfaces/publish.interface';

export const CRenderItem = ({title, body, userId}: IPublication) => {
  return (
    <View accessibilityRole='menuitem' style={styles.container}>
      <Text style={styles.title}>{title}</Text>
      <Text>{body}</Text>
      <Text>User: {userId}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: 10,
    borderBottomColor: '#8BA0AB',
    borderBottomWidth: 1,
  },
  title: {
    fontWeight: 'bold',
    fontSize: 20,
  },
});

CRenderItem.propTypes = {
  item: Proptypes.shape({
    id: Proptypes.number.isRequired,
    title: Proptypes.string.isRequired,
    body: Proptypes.string.isRequired,
    userId: Proptypes.number.isRequired,
  }),
};
