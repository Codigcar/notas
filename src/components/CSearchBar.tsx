import React from 'react';
import {View, Platform} from 'react-native';
import {SearchBar} from 'react-native-elements';
import {SearchBarBaseProps} from 'react-native-elements/dist/searchbar/SearchBar';
import PropTypes from 'prop-types';

 export const CSearchBar = ({
  dataList,
  search,
  setSearch,
  setFilteredDataSource,
}: any) => {
  const SafeSearchBar = SearchBar as unknown as React.FC<SearchBarBaseProps>;

  const searchFilterFunction = (text: string) => {
    if (text) {
      const newData = dataList.filter(function (item: any) {
        const itemData = item.title;
        return itemData.indexOf(text) > -1;
      });
      setFilteredDataSource(newData);
      setSearch(text);
    } else {
      setFilteredDataSource(dataList);
      setSearch(text);
    }
  };

  return (
    <View>
      <SafeSearchBar
        containerStyle={styles.containerBarraBusqueda}
        inputContainerStyle={styles.estiloBarraBusqueda}
        onChangeText={(text: string) => searchFilterFunction(text)}
        onClear={() => searchFilterFunction('')}
        placeholder="Buscar por tÃ­tulo..."
        value={search}
        platform={'android'}
        clearIcon={true}
        testID={'search'}
      />
    </View>
  );
};

export const styles = {
  containerBarraBusqueda: {
    backgroundColor: '#fff',
    borderTopColor: '#FFF',
    borderBottomColor: '#FFF',
    paddingHorizontal: 0,
    marginRight: 20,
    ...Platform.select({
      ios: {
        paddingVertical: 0,
        borderRadius: 10,
      },
    }),
  },
  estiloBarraBusqueda: {
    backgroundColor: 'white',
    borderRadius: 10,
    marginVertical: 0,
    ...Platform.select({
      ios: {},
      android: {
        marginLeft: 15,
        marginRight: 15,
        shadowOpacity: 0.39,
        shadowRadius: 13.97,
        elevation: 11,
      },
      default: {
        shadowColor: 'rgba(0,0,0, .2)',
        shadowOffset: {height: 0, width: 0},
        shadowOpacity: 1,
        shadowRadius: 1,
      },
    }),
  },
};

CSearchBar.propTypes = {
  dataList: PropTypes.array.isRequired,
  search: PropTypes.string.isRequired,
  setSearch: PropTypes.func.isRequired,
  setFilteredDataSource: PropTypes.func.isRequired,
}
