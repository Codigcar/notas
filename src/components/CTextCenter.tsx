import React from 'react';
import {View, Text, StyleSheet} from 'react-native';
import Proptypes from 'prop-types';

type Props = {
  title: string;
};

export const CTextCenter = ({title}: Props) => {
  return (
    <View style={styles.center}>
      <View>
        <Text>{title}</Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  center: {
    justifyContent: 'center',
    alignItems: 'center',
    flex: 1,
  },
});

CTextCenter.propTypes = {
    title: Proptypes.string.isRequired
}