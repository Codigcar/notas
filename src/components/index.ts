

export * from './CRenderItem';
export * from './CRenderList';
export * from './CSearchBar';
export * from './CTextCenter';