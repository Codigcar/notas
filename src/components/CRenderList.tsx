import React from 'react';
import {FlatList, RefreshControl } from 'react-native';
import {CRenderItem} from './CRenderItem';
import PropTypes from 'prop-types';
import {CTextCenter} from './CTextCenter';

type Props = {
  data: any;
  pullRefresh: () => Promise<void>;
  isLoadingPullRefresh: boolean;
  loadMore?: (() => void) | null;
  timeEndReached?: number | null;
};

export const CRenderList = ({
  data,
  loadMore = null,
  timeEndReached = null,
  pullRefresh,
  isLoadingPullRefresh,
}: Props) => {
  return (
    <FlatList
      testID='flat-list'
      data={data}
      keyExtractor={item => item.id.toString()}
      renderItem={({item}) => <CRenderItem {...item} />}
      onEndReached={loadMore}
      onEndReachedThreshold={timeEndReached}
      refreshControl={
        <RefreshControl
          refreshing={isLoadingPullRefresh}
          onRefresh={pullRefresh}
          tintColor={'red'}
          colors={['red']}
          testID={'refresh-control'}
        />
      }
      contentContainerStyle={{flexGrow: 1}}
      ListEmptyComponent={() => {
        return <CTextCenter title="Publicaciones no encontradas" />;
      }}
    />
  );
};

CRenderList.propTypes = {
  data: PropTypes.array.isRequired,
  pullRefresh: PropTypes.func.isRequired,
  isLoadingPullRefresh: PropTypes.bool.isRequired,
  loadMore: PropTypes.func,
  timeEndReached: PropTypes.number,
};
