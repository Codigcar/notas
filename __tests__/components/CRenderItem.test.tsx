import React from 'react';
import { CRenderItem } from '../../src/components/CRenderItem';

const { render , screen} = require('@testing-library/react-native')

describe('testing en <CRenderItem />', () => {
    const item = {
        id: 1,
        title:'title',
        body:'body',
        userId:1
    }
    it('debe hacer match con el snapshot', () => {
        const component = render(<CRenderItem {...item} />)
        expect(component).toMatchSnapshot()
    });

    it('debe renderizar los textos', () => {

        const {getByText} = render(<CRenderItem {...item} />)
        
        expect(getByText(item.title)).toBeTruthy();
        expect(getByText(item.body)).toBeTruthy();
        expect(getByText(`User: ${item.userId}`)).toBeTruthy();
    });
});