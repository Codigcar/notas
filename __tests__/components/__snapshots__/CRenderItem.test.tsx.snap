// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing en <CRenderItem /> debe hacer match con el snapshot 1`] = `
<View
  accessibilityRole="menuitem"
  style={
    Object {
      "borderBottomColor": "#8BA0AB",
      "borderBottomWidth": 1,
      "padding": 10,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    title
  </Text>
  <Text>
    body
  </Text>
  <Text>
    User: 
    1
  </Text>
</View>
`;
